src/Components/Button/ButtonComponent.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from "react";
+import { ButtonProps } from "./types";
+import { getButtonStyles } from "./styles";
+import Button from "@mui/material/Button";
+
+const ButtonComponent = ({
+  label,
+  type,
+  size,
+  disabled,
+  onClick,
+}: ButtonProps) => {
+  const [isPressed, setIsPressed] = useState(false);
+
+  const handlePress = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setIsPressed(!isPressed);
```

Comment:
change it to `setIsPressed(prevState => !prevState)`.
We'll talk tomorrow about what is the difference.